internal/controller/components/values.go

```diff
@@ -92,26 +92,57 @@ func getComponentValues(
 		}
 
 	case kcmv1.ProviderSveltosName:
-		componentValues = map[string]any{
-			"projectsveltos": map[string]any{
-				"registerMgmtClusterJob": map[string]any{
-					"registerMgmtCluster": map[string]any{
-						"args": []string{
-							"--labels=" + kcmv1.K0rdentManagementClusterLabelKey + "=" + kcmv1.K0rdentManagementClusterLabelValue,
-						},
-					},
+		projectsveltos := make(map[string]any)
+		projectsveltos["registerMgmtClusterJob"] = map[string]any{
+			"registerMgmtCluster": map[string]any{
+				"args": []string{
+					"--labels=" + kcmv1.K0rdentManagementClusterLabelKey + "=" + kcmv1.K0rdentManagementClusterLabelValue,
 				},
 			},
 		}
+
+		if opts.ImagePullSecretName != "" {
+			imagePatch := `- op: add
+  path: /spec/template/spec/imagePullSecrets
+  value:
+  - name: ` + opts.ImagePullSecretName
+
+			projectsveltos["classifierManager"] = map[string]any{
+				"agentPatchConfigMap": map[string]any{
+					"data": map[string]any{
+						"image-patch": imagePatch,
+					},
+				},
+			}
+		}
+
+		componentValues = map[string]any{
+			"projectsveltos": projectsveltos,
+		}
 	}
 
 	if opts.GlobalRegistry != "" {
-		globalValues := map[string]any{
-			"global": map[string]any{
-				"registry": opts.GlobalRegistry,
-			},
+		vals := make(map[string]any)
+		global := map[string]any{
+			"registry": opts.GlobalRegistry,
 		}
-		componentValues = chartutil.CoalesceTables(componentValues, globalValues)
+
+		if opts.ImagePullSecretName != "" {
+			global["imagePullSecrets"] = []map[string]any{
+				{
+					"name": opts.ImagePullSecretName,
+				},
+			}
+
+			vals["configSecret"] = map[string]any{
+				"create": false,
```

Comment:
I can't find this field in the `InfrastructureProvider` spec, could you elaborate where the field comes from?